spaceship/lib/spaceship/tunes/tunes_client.rb

```diff
@@ -605,6 +605,17 @@ def update_availability!(app_id, availability)
       data["countries"] = availability.territories.map { |territory| { 'code' => territory.code } }
       data["theWorld"] = availability.include_future_territories.nil? ? true : availability.include_future_territories
 
+      # InitializespreOrder (if needed)
+      data["preOrder"] ||= {}
+
+      # Sets app_available_date to nil if cleared_for_preorder if false
+      # This is need for apps that have never set either of these before
+      # API will error out if cleared_for_preorder is false and app_available_date has a date
+      cleared_for_preorder = availability.cleared_for_preorder
+      app_available_date = cleared_for_preorder ? availability.app_available_date : nil
+      data["preOrder"]["clearedForPreOrder"] = { "value" => cleared_for_preorder, "isEditable" => true, "isRequired" => true, "errorKeys" => nil }
```

Comment:
I believe if you use the `attr_mapping` you don't make to create these hashes, they already exist on the model in this format.